Automattic/simplenote-ios
# New note widget

### Fix
This PR is the first part of the iOS widget project #1289 

This is PR add the basic infrastructure to add widgets into Simplenote (adding widget target, widget group etc)

It also includes the New Note Widget.  This widget is a home screen button to quickly jump straight into the new note screen.

<img src="https://cdn-std.droplr.net/files/acc_593859/zm1YV7" />

The widget it's self is pretty simple, it has no timeline, and displays no details.  It simply triggers opening the app.

### Test
1. Build the app on your device/simulator
2. Go to the home screen and press and hold to make the homepage edit appear, tap the plus button
3. Find Simplenote and add the new note widget
4. tap on it, you should be taken to new note page

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

Also, @mokagio  would you please take a look at this PR to see if there are any changes to the SimplenoteWidgetExtension configuration that we should make.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in 216f7d with:
> 
> > Adds a home screen widget to start a new note with a single tap
